src/mito2/src/sst/parquet/file_range.rs

```diff
@@ -144,6 +173,34 @@ impl FileRangeContext {
     pub(crate) fn precise_filter(&self, input: Batch) -> Result<Option<Batch>> {
         self.base.precise_filter(input)
     }
+
+    //// Decodes parquet metadata and finds if row group contains delete op.
+    pub(crate) fn contains_delete(&self, row_group_index: usize) -> Result<bool> {
```

Comment:
non-blocking: We can change the return value to `Option<bool>` and return None if we can't get the stat.